packages/odata-react/src/store/index.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+import createUseContext from "constate";
+
+interface StoreShape {
+  [key: string]: boolean;
+}
+
+const useCollapseStateRaw = () => {
+  const [state, setState] = useState<StoreShape>({});
+  const updatePart = (arg: StoreShape) => setState({ ...state, ...arg });
```

Comment:
Please change name of this function and also for `updateWholeState`. For my naming like `updateState` etc by related to specific part of state is very unreadable to 3rd person, so `updatePartOfCollapseState` would be better.